src/v/kafka/server/audit_log_manager.cc

```diff
@@ -0,0 +1,569 @@
+        } catch (...) {
+            /// Sleep, then try again
+        }
+        co_await ss::sleep_abortable(1s).handle_exception_type(
+          [](ss::sleep_aborted) {});
+    }
+}
+
+ss::future<> audit_client::configure() {
+    try {
+        auto config = co_await client::create_client_credentials(
+          *_controller,
+          config::shard_local_cfg(),
+          _client.config(),
+          audit_principal);
+        set_client_credentials(*config, _client);
+
+        auto const& store
+          = _controller->get_ephemeral_credential_store().local();
+        _has_ephemeral_credentials = store.has(store.find(audit_principal));
+
+        co_await set_auditing_permissions();
+        co_await create_internal_topic();
+
+        _client.config().retries.set_value(std::numeric_limits<size_t>::max());
+        vlog(klog.info, "Audit log subsystem initialized");
+    } catch (...) {
+        vlog(
+          klog.warn,
+          "Audit log subsystem failed to initialize: {}",
+          std::current_exception());
+        throw;
+    }
+}
+
+ss::future<> audit_client::set_auditing_permissions() {
+    /// Give permissions to create and write to the audit topic
+    security::acl_entry acl_create_entry{
+      audit_principal,
+      security::acl_host::wildcard_host(),
+      security::acl_operation::create,
+      security::acl_permission::allow};
+
+    security::acl_entry acl_write_entry{
+      audit_principal,
+      security::acl_host::wildcard_host(),
+      security::acl_operation::write,
+      security::acl_permission::allow};
+
+    security::resource_pattern audit_topic_pattern{
+      security::resource_type::topic,
+      model::kafka_audit_logging_topic,
+      security::pattern_type::literal};
+
+    /// TODO: Add rules for User:* deny to things like deleting the topic
+    /// and other unwanted behavior. User:* should be allowed to ::describe
+    /// and alter topic configs
+    co_await _controller->get_security_frontend().local().create_acls(
+      {security::acl_binding{audit_topic_pattern, acl_create_entry},
+       security::acl_binding{audit_topic_pattern, acl_write_entry}},
+      5s);
+}
+
+ss::future<> audit_client::mitigate_error(std::exception_ptr eptr) {
+    if (_gate.is_closed() || _as.abort_requested()) {
+        /// TODO: Investigate looping behavior on shutdown
+        co_return;
+    }
+    vlog(klog.trace, "mitigate_error: {}", eptr);
+    auto f = ss::now();
+    try {
+        std::rethrow_exception(eptr);
+    } catch (client::broker_error const& ex) {
+        if (
+          ex.error == error_code::sasl_authentication_failed
+          && _has_ephemeral_credentials) {
+            f = inform(ex.node_id).then([this]() { return _client.connect(); });
+        } else {
+            throw;
+        }
+    } catch (...) {
+        throw;
+    }
+    co_await std::move(f);
+}
+
+ss::future<> audit_client::inform(model::node_id id) {
+    vlog(klog.trace, "inform: {}", id);
```

Comment:
suggestion: I think audit should have it's own logging facility

Reply:
For now this lives in `src/v/kafka` until we perform the move